launcher/minecraft/mod/ModFolderModel_test.cpp

```diff
@@ -32,17 +66,23 @@ private
         {
             QString folder = source;
             QTemporaryDir tempDir;
-            ModFolderModel m(tempDir.path());
+            QEventLoop loop;
+            ModFolderModel m(tempDir.path(), true);
+            connect(&m, &ModFolderModel::updateFinished, &loop, &QEventLoop::quit);
             m.installMod(folder);
+            loop.exec();
```

Comment:
i think the race condition still exists here (and below), albeit with a smaller time window. It can happen that the scheduler might make it so that the ModFolderModel finishes updating _before_ the event loop actually starts executing, and the signal is lost, so it remains in the loop forever.

It's a small possibility though, so I don't think there's much problem leaving it as-is. You decide :)

Reply:
That is true. But I think there is no other way to do this. All code examples from Qt work this way as well. So we are kinda out of luck here :(

Reply:
In theory you could set up a timer that periodically exits the loop, and put those two inside a `while (not_finished)` or something like that, with the `not_finished` being set in a lambda that gets called by the `updateFinished` signal. Probably too much work for this slim chance of an error, so yea, we're out of luck ;(